webapp-django/crashstats/status/context_processors.py

```diff
@@ -2,19 +2,22 @@
 
 
 def status_message(request):
-    statuses = StatusMessage.objects.filter(enabled=True)
 
-    if not statuses.exists():
-        return {}
+    def inner():
+        statuses = StatusMessage.objects.filter(enabled=True)
 
-    messages = []
-    for status in statuses.order_by('-created_at'):
-        messages.append({
-            'text': status.message,
-            'severity': status.severity,
-            'date': status.created_at,
-        })
+        if not statuses.exists():
+            return {}
+
+        messages = []
+        for status in statuses.order_by('-created_at'):
+            messages.append({
+                'text': status.message,
+                'severity': status.severity,
+                'date': status.created_at,
+            })
+        return messages
```

Comment:
Seems like we really want to call `statuses.order_by('-created_at').values('message', 'severity', 'created_at')` and skip this for loop. Then the SQL query is only performed if it's used in the template and you don't need to create an `inner()` function that only gets called in the template and all that.

Does that sound right?